tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -806,6 +809,24 @@ impl<'a> fmt::Debug for DefaultVisitor<'a> {
     }
 }
 
+/// Renders an error into a list of sources, *including* the error
+struct ErrorSourceList<'a>(&'a (dyn std::error::Error + 'static));
+
+impl<'a> Display for ErrorSourceList<'a> {
```

Comment:
another thought i had, but again, take it or leave it: i wonder if a nicer way to factor this would be to have this type just implement `Iterator`, analogous to the [`Error::chain`](https://doc.rust-lang.org/stable/std/error/struct.Chain.html) unstable API. then, we could have the default formatter use `f.debug_list().entries(ErrorSourceList(&error)).finish()`, but the `Pretty` formatter could do something arbitrarily fancy with the iterator, like delimiting the errors with indentation...and eventually, we could also use the iterator in the JSON formatter to serialize the error chain as a JSON list.

what do you think? i'd be fine with merging this as-is, and making that change later if and when it's necessary, but it seems like it might be a somewhat nicer way to factor this?

Reply:
> then, we could have the default formatter

The `Display` of `ErrorSourceList`?